aea/protocols/generator/base.py

```diff
@@ -183,24 +189,50 @@ class ProtocolGenerator:
 
     def __init__(
         self,
-        protocol_specification: ProtocolSpecification,
+        path_to_protocol_specification: str,
         output_path: str = ".",
         path_to_protocol_package: Optional[str] = None,
     ) -> None:
         """
         Instantiate a protocol generator.
 
-        :param protocol_specification: the protocol specification object
+        :param path_to_protocol_specification: path to protocol specification file
         :param output_path: the path to the location in which the protocol module is to be generated.
         :param path_to_protocol_package: the path to the protocol package
 
         :return: None
         """
-        self.protocol_specification = protocol_specification
+        self.path_to_protocol_specification = path_to_protocol_specification
```

Comment:
Now this method does even more; please isolate functionalities...e.g. have individual methods which wrap separate operations and then call them from here. Some of these are not related to the object so they can be helper methods

Reply:
Done